rlclientlib/logger/http_transport_client.h

```diff
@@ -174,6 +179,9 @@ namespace reinforcement_learning {
   template <typename TAuthorization>
   int http_transport_client<TAuthorization>::init(const utility::configuration& config, api_status* status) {
     RETURN_IF_FAIL(_authorization.init(config, status, _trace));
+    _config = new utility::configuration(config);
```

Comment:
do we need to create a new configuration object here just so that we can get a setting? could we not just re-use the `config` to do `const char* section = config->get(config_constants::CONFIG_SECTION, config_constants::INTERACTION);`?

Reply:
we need the config when we are trying to create new HTTP client with the new URL (in case we are appending events_counts in the URL and creating new HTTP client)

Reply:
If the new client is passed into the `http_transport_client` constructor same as the current `_client` then it can be constructed in `azure_factories.cc` where the config is known. That way we don't need to create a new config object and store it here

Reply:
connects with this comment: https://github.com/VowpalWabbit/reinforcement_learning/pull/447/files/b08410ab9bc18a301d2939ec68f7cdd37fe11501#r807140690

Reply:
if the configuration needs to be copied here it should be a unique_ptr so that its memory is cleaned up during destruction